connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1150,6 +1198,34 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
+  test("Corner case: retain empty recursive proto fields") {
+    // This verifies that a empty proto like 'message A { A a = 1}' can be retained by
+    // inserting a dumy field.
+
+    val emptyProtoSchema =
+      StructType(StructField("_dummy_field_to_retain_empty_message", StringType) :: Nil)
+    val expectedSchema = StructType(
+      // DDL: "proto STRUCT<name: string, groups: map<
+      //    struct<name: string, group: map<struct<name: string>>>>>"
+      StructField("empty_proto",
+        StructType(
+          StructField("recursive_field", emptyProtoSchema) ::
+            StructField("recursive_array", ArrayType(emptyProtoSchema, containsNull = false)
```

Comment:
Where is the dummy field here?

Reply:
Inside emptyProtoSchema

Reply:
Could you update the DDL comment at 1208?